src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -197,6 +202,77 @@ public decimal GetBalance(string accountName, int minConfirmations = 0)
             return balance?.ToUnit(MoneyUnit.BTC) ?? 0;
         }
 
+        [ActionName("listsinceblock")]
+        [ActionDescription("Get all transactions in blocks since block 'blockhash', or all transactions if omitted.")]
+        public async Task<ListSinceBlockModel> ListSinceBlockAsync(string blockHash, int targetConfirmations = 1)
+        {
+            WalletAccountReference accountReference = this.GetWalletAccountReference();
+            Wallet wallet = this.walletManager.GetWallet(accountReference.WalletName);
+            Block block = null;
+
+            if (!string.IsNullOrEmpty(blockHash) && uint256.TryParse(blockHash, out uint256 hashBlock))
+            {
+                this.ConsensusManager.GetOrDownloadBlocks(new List<uint256> { hashBlock }, b => { block = b.Block; });
+            }
+
+            IEnumerable<TransactionData> transactions = wallet.GetAllTransactions();
+
+            var model = new ListSinceBlockModel();
+
+            foreach (TransactionData transactionData in transactions)
+            {
+                GetTransactionModel transaction = await this.GetTransactionAsync(transactionData.Id.ToString());
+
+                int blockHeight = transactionData.BlockHeight ?? 0;
+
+                if (block != null && blockHeight < this.ChainIndexer.GetHeader(block?.GetHash()).Height)
+                    continue;
+
+                if (transaction.Confirmations < targetConfirmations)
+                    continue;
+
+                foreach (GetTransactionDetailsModel transactionDetail in transaction.Details)
+                {
+                    ListSinceBlockTransactionCategoryModel category = GetListSinceBlockTransactionCategoryModel(transaction);
+
+                    model.Transactions.Add(new ListSinceBlockTransactionModel
+                    {
+                        Confirmations = transaction.Confirmations,
+                        BlockHash = transaction.BlockHash,
+                        BlockIndex = transaction.BlockIndex,
+                        BlockTime = transaction.BlockTime,
+                        TransactionId = transaction.TransactionId,
+                        TransactionTime = transaction.TransactionTime,
+                        TimeReceived = transaction.TimeReceived,
+                        Account = accountReference.AccountName,
+                        Address = transactionDetail.Address,
+                        Amount = transactionDetail.Amount,
+                        Category = category,
+                        Fee = transaction.Fee,
+                        OutputIndex = transactionDetail.OutputIndex
+                    });
+                }
+            }
+
+            model.LastBlock = this.ConsensusManager.Tip.HashBlock;
```

Comment:
It's better practice to use the tip of ChainIndexer

Reply:
Resolved in commit 4753e125151cd2df0bb6badaeef9b1392e7855c7